packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -1090,11 +1102,44 @@ class PopupMenuButton<T> extends StatefulWidget {
   PopupMenuButtonState<T> createState() => PopupMenuButtonState<T>();
 }
 
+/// Signature for the callback used by [showMenu] to obtain the position of the
+/// [PopupMenuButton].
+///
+/// Used by [showMenu].
+typedef PopupMenuButtonPositionCallback = RelativeRect Function();
+
 /// The [State] for a [PopupMenuButton].
 ///
 /// See [showButtonMenu] for a way to programmatically open the popup menu
 /// of your button state.
 class PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
+  GlobalKey _menuGlobalKey = GlobalKey();
+  RelativeRect? _buttonPosition;
+
+  RelativeRect _getButtonPosition() => _buttonPosition!;
+
+  RelativeRect _calculateButtonPosition() {
+    final RenderBox button = context.findRenderObject()! as RenderBox;
+    final RenderBox overlay = Navigator.of(context).overlay!.context.findRenderObject()! as RenderBox;
+    return RelativeRect.fromRect(
+      Rect.fromPoints(
+        button.localToGlobal(widget.offset, ancestor: overlay),
+        button.localToGlobal(button.size.bottomRight(Offset.zero) + widget.offset, ancestor: overlay),
+      ),
+      Offset.zero & overlay.size,
+    );
+  }
+
+  void _maybeUpdateMenuPosition() {
+    WidgetsBinding.instance!.addPostFrameCallback((Duration duration) {
```

Comment:
Ahh, I guess the delay comes from it being a post-frame callback. It only attempts to update the button position a frame after. I tried to look at what happens with the GitHub UI on the web and it's pretty interesting -- The movement of the entire window itself stalls until the UI is able to draw itself to the new intended screen size.

This will cause the behavior to differ from what happens natively, but it has its own benefit in that the resizing of the window doesn't stutter, but the menu position lags behind.

cc/ @HansMuller for his thoughts on maybe other ways we can tackle this

Pay attention to the location of the cursor relative to the position of the window (this is showing native behavior for web):


https://user-images.githubusercontent.com/27032613/114829066-38ca4980-9dfd-11eb-99fd-7a13ce6c4336.mov

Reply:
We don't usually do things like this (using a post frame callback, etc). If the overlay that the menu is parented to changes size after the menu has appeared, maybe we should just dismiss the menu?

Reply:
Is there a specification for this behavior? Now it seems that this is not done on the web.

Reply:
I don't believe there's a Material Design spec for this behavior and it's true that in some cases web apps will endeavor to keep a popup menu up when the containing window is moved or resized.  On the other hand, most desktop toolkits and apps do dismiss popups when the user changes the containing window's geometry or visibility. I think it makes sense to dismiss the menu in this case, since - from the user's perspective - selecting an item from a popup menu is a gesture with a short lifetime. Apps that want to give the user a more persistent UI for selecting an option or starting an action, usually use dialogs or something else that the user would expect to hang around.

Reply:
![20210421_084633](https://user-images.githubusercontent.com/61075224/115480987-32fdb980-a27e-11eb-9eef-0b9ad1207ee2.gif)

Reply:
I have tested this behavior on Android native.